model/lists/ListService.php

```diff
@@ -15,46 +15,40 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2018 (original work) Open Assessment Technologies SA;
- *
- *
+ * Copyright (c) 2018-2021 (original work) Open Assessment Technologies SA;
  */
 
+declare(strict_types=1);
+
 namespace oat\taoBackOffice\model\lists;
 
+use tao_models_classes_ListService;
+use Psr\Container\ContainerInterface;
 use core_kernel_classes_Class as RdfClass;
-use core_kernel_persistence_Exception;
 use oat\generis\model\kernel\uri\UriProvider;
 use oat\tao\model\Lists\Business\Domain\Value;
 use oat\tao\model\Lists\Business\Domain\ValueCollection;
-use oat\tao\model\Lists\Business\Domain\ValueCollectionSearchRequest;
+use oat\tao\model\Specification\ClassSpecificationInterface;
+use oat\tao\model\Lists\Business\Service\ValueCollectionService;
 use oat\tao\model\Lists\Business\Input\ValueCollectionDeleteInput;
 use oat\tao\model\Lists\Business\Input\ValueCollectionSearchInput;
-use oat\tao\model\Lists\Business\Service\RemoteSourcedListOntology;
-use oat\tao\model\Lists\Business\Service\ValueCollectionService;
+use oat\tao\model\Lists\Business\Domain\ValueCollectionSearchRequest;
+use oat\tao\model\Lists\Business\Specification\RemoteListClassSpecification;
 use oat\tao\model\Lists\DataAccess\Repository\ParentPropertyListCachedRepository;
-use oat\tao\model\TaoOntology;
-use tao_models_classes_LanguageService;
-use tao_models_classes_ListService;
 
-/**
- * Class ListService
- */
 class ListService extends tao_models_classes_ListService
 {
     /**
+     * @TODO Make two different kind of lists: system list that are not editable and usual list.
```

Comment:
I would remove this TODO, does not make sense anymore and this is a business requirement.